rust/src/krb/krb5.rs

```diff
@@ -111,13 +114,26 @@ impl KRB5State {
                 if hdr.class != BerClass::Application { return 0; }
                 match hdr.tag.0 {
                     10 => {
+                        let req = krb5_parser::parse_as_req(i);
+                        if let Ok((_,kdc_req)) = req {
+                            let mut tx = self.new_tx();
+                            tx.msg_type = MessageType::KRB_AS_REQ;
+                            tx.cname = kdc_req.req_body.cname;
+                            tx.realm = Some(kdc_req.req_body.realm);
+                            tx.sname = kdc_req.req_body.sname;
+                            tx.etype = None;
+                            self.transactions.push(tx);
+                        };
                         self.req_id = 10;
                     },
                     11 => {
                         let res = krb5_parser::parse_as_rep(i);
                         if let Ok((_,kdc_rep)) = res {
                             let mut tx = self.new_tx();
                             tx.msg_type = MessageType::KRB_AS_REP;
+                            if self.req_id > 0 {
+                                tx.req_type = Some(MessageType(self.req_id.into()));
```

Comment:
Could you please add a comment about this ?
Why using the condition `if self.req_id > 0`
My understanding is that what got parsed as a request id, is not in this case a request id but a request type...
Why did we parse it as a request id in the first place ?

Reply:
Hi,

You are totally correct, the req_id is the request type, however that was the name of the variable in the code. I wanted to do as few changes in the code as I could so I keep this variable name (I also needed to introduce a field to keep track of request_type in responses, so I already use the word `req_type`, more accurate). But yeah, it requires a rename, 
 probably in a improvement PR.

Related to the condition `if self.req_id > 0`: in the code self.req_id is only set different to zero for requests, so you only get the request type when the previous one was a request, in other case, you keep the default None Option.

Reply:
Thanks for the explanation. This naming deserves a comment

Reply:
Done in https://github.com/OISF/suricata/pull/6508/commits/abbef13b12b58e8b6c2fb7fff6b589d97cfacde5